src/storage/src/storage_state/async_storage_worker.rs

```diff
@@ -92,11 +184,73 @@ impl<T: Timestamp + Lattice + Codec64> AsyncStorageWorker<T> {
                             .calculate_resumption_frontier(&mut state)
                             .instrument(span)
                             .await;
+
+                        // Create a specialized description to be able to call the generic method
+                        let source_resume_upper = match ingestion_description.desc.connection {
+                            GenericSourceConnection::Kafka(_) => {
```

Comment:
I'm fine with us needing this match statement, but I think it should be broken out into a `source::reclock` submodule